File: src/main/java/org/visual/component/GlobalSystemMenu.java
package org.visual.component;

import java.util.function.Supplier;
import javafx.scene.control.MenuItem;
import javafx.scene.input.KeyCombination;
import lombok.val;
import org.visual.component.display.SystemMenuBar;

public class GlobalSystemMenu extends SystemMenuBar {

  public GlobalSystemMenu() {
    setUseSystemMenuBar(true);
  }

  private final Supplier<MenuItem> exitItem =
      () -> {
        val menuItem = new MenuItem("exit");
        menuItem.acceleratorProperty().set(KeyCombination.keyCombination("CTRL+Q"));
        menuItem.setOnAction(actionEvent -> {});
        return menuItem;
      };
}

File: src/main/kotlin/org/visual/component/GlobalSystemMenu.kt
package org.visual.component

import jakarta.inject.Singleton
import javafx.scene.control.Menu
import javafx.scene.control.MenuItem
import javafx.scene.input.KeyCombination
import javafx.stage.Stage
import org.visual.component.display.SystemMenuBar

@Singleton
class GlobalSystemMenu : SystemMenuBar() {

  private val exitItem by lazy {
    MenuItem("Exit").apply {
      accelerator = KeyCombination.keyCombination("CTRL+Q")
      setOnAction {
        val stage = scene.window as Stage
        stage.close()
      }
    }
  }

  private val fileMenu by lazy { Menu("File").apply { items.add(exitItem) } }

  init {
    menus.add(fileMenu)
  }
}
